pkg/services/ngalert/notifier/status.go

```diff
@@ -1,16 +1,19 @@
 package notifier
 
 import (
+	"encoding/json"
+
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 )
 
+// TODO: We no longer do apimodels at this layer, move it to the API.
 func (am *Alertmanager) GetStatus() apimodels.GettableStatus {
-	am.reloadConfigMtx.RLock()
-	defer am.reloadConfigMtx.RUnlock()
-
-	config := apimodels.PostableApiAlertingConfig{}
-	if am.ready() {
-		config = am.config.AlertmanagerConfig
+	config := &apimodels.PostableUserConfig{}
+	if am.Base.Ready() {
+		if err := json.Unmarshal(am.Base.GetStatus(), config); err != nil {
```

Comment:
Do we still need the lock here? there is a gap between `am.Base.Ready()` and `am.Base.GetStatus()`, the config could be concurrently updated there.

Reply:
Good shout -- it was a bit confusing, so I've removed the `am.Base.Ready()` check.

If you look at `Base.GetStatus` it does hold a lock for getting the status from the configuration - so really all that we need is to execute the call against this check and safely remove the check to `Base.Ready()` as you correctly guessed it could change between those two calls so we don't gain anything.

Reply:
So to answer your question concretely  - no, we don't need a lock and we also don't need the `Ready` check as we can rely on `Base.GetStatus` to give us the accurate information.